src/legacy/server/core/index.ts

```diff
@@ -21,10 +21,20 @@ import { Server } from 'hapi';
 import KbnServer from '../kbn_server';
 
 /**
- * Exposes `kbnServer.core` through Hapi API.
+ * Exposes `kbnServer.newPlatform` through Hapi API.
  * @param kbnServer KbnServer singleton instance.
  * @param server Hapi server instance to expose `core` on.
  */
 export function coreMixin(kbnServer: KbnServer, server: Server) {
-  server.decorate('server', 'core', kbnServer.core);
```

Comment:
to avoid confusion I removed this escape hatch, let me know if we want to keep it and revert the commit